### PR TITLE
JAVA-2581: Add query builder support for indexed list assignments

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

File: query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/update/Assignment.java [hunk 1]
```diff
@@ -87,8 +87,25 @@ static Assignment setMapValue(
...
   static Assignment setMapValue(
+      @NonNull String columnName, @NonNull Term key, @NonNull Term value) {
+    return setMapValue(CqlIdentifier.fromCql(columnName), key, value);
+  }
+
+  /** Assigns a value to an index in a list column, as in {@code SET list[?]=?}. */
+  @NonNull
+  static Assignment setListValue(
+      @NonNull CqlIdentifier columnId, @NonNull Term index, @NonNull Term value) {
+    return new DefaultAssignment(new ColumnComponentLeftOperand(columnId, index), "=", value);
```
Comment (adutra) [inline query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/update/Assignment.java:99]: The syntax for lists is the same for maps.

File: query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/update/UpdateFluentAssignmentTest.java [hunk 2]
```diff
@@ -60,6 +60,16 @@ public void should_generate_map_value_assignment() {
...
 
+  @Test
+  public void should_generate_list_value_assignment() {
+    assertThat(
+            update("foo")
+                .setListValue("features", literal(1), bindMarker())
+                .whereColumn("k")
+                .isEqualTo(bindMarker()))
+        .hasCql("UPDATE foo SET features[1]=? WHERE k=?");
+  }
```
Comment (tomekl007) [inline query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/update/UpdateFluentAssignmentTest.java:71]: missing test for map

Comment (adutra) [reply query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/update/UpdateFluentAssignmentTest.java:71]: it's just above this test on line 64: `should_generate_map_value_assignment`.

**Review by tomekl007 — APPROVED** [review]

**Review by olim7t — APPROVED** [review]
Yeah the intent was that `setMapValue` could be used for both, but OK to add synonyms for clarity.

**Pull request merged**